### PR TITLE
Core/Scripts: implement warrior spell Storm Bolt

### PR DESCRIPTION
**Pull request by HannibalRoG**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  Implement Warrior spell Storm Bolt (stun).

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
Compiled and tested in game.

### CODE REVIEW

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 1]
```diff
@@ -683,6 +685,45 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
...
+                SPELL_WARRIOR_STORM_BOLT_STUN
+            });
+        }
+
+        void HandleDamage(SpellEffIndex /*effIndex*/)
+        {
+            Unit* caster = GetCaster();
+
+            if (Unit* target = GetHitUnit())
+                caster->CastSpell(target, SPELL_WARRIOR_STORM_BOLT_STUN, true);
```
Comment (Temppus) [inline src/server/scripts/Spells/spell_warrior.cpp]: Dont need to store caster pointer. Just GetCaster()->CastSpell(.....);

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 2]
```diff
@@ -53,6 +53,8 @@ enum WarriorSpells
     SPELL_WARRIOR_SECOUND_WIND_TRIGGER_RANK_2       = 29842,
     SPELL_WARRIOR_SHIELD_SLAM                       = 23922,
     SPELL_WARRIOR_SLAM                              = 50782,
+    SPELL_WARRIOR_STORM_BOLT                        = 107570,
```
Comment (DoctorKraft) [inline src/server/scripts/Spells/spell_warrior.cpp]: unused

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 3]
```diff
@@ -683,6 +685,45 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
...
+
+    class spell_warr_storm_bolt_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_warr_storm_bolt_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_STORM_BOLT,
```
Comment (DoctorKraft) [inline src/server/scripts/Spells/spell_warrior.cpp]: same unused

**Comment by DoctorKraft** [comment]
Wrong effect also, no ? You need add this in `SPELL_EFFECT_DUMMY` (Effect1) with `OnEffectHitTarget`

**Comment by P-Kito** [comment]
SPELL_EFFECT_DUMMY -> OnEffectHitTarget -> Cast 132169

Make sure that the stormbolt flys to the target before stunning it (small delay if it flys a long range)

**Comment by HannibalRoG** [comment]
Now when the day of love passes we review and fix it, by the way, congratulations to all by February 14th.

**Comment by Ascathor** [comment]
@HannibalRoG Same right back to you, even if it's not a celebration to me. However, it is well hoped that you had a fine day and enjoyed it with your loved one.

File: src/server/scripts/Spells/spell_warrior.cpp (hunk 3 above)
Comment (Shauren) [inline src/server/scripts/Spells/spell_warrior.cpp]: Don't validate the spell you are attaching the script to (107570), it is done by ObjectMgr already and the script is never created

**Comment by HannibalRoG** [comment]
Any comments now?

**Comment by tkrokli** [comment]
Maybe you could reply to Temppus in his comment https://github.com/TrinityCore/TrinityCore/pull/19130#pullrequestreview-21626837. I don't know if his suggestion is correct or not.

File: src/server/scripts/Spells/spell_warrior.cpp (hunk 1 above)
Comment (HannibalRoG) [inline src/server/scripts/Spells/spell_warrior.cpp]: Ups, I passed this comment, thanks @tkrolki.
We fix it soon.

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 4]
```diff
@@ -683,6 +684,43 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
...
+        {
+            return ValidateSpellInfo
+            ({
+                SPELL_WARRIOR_STORM_BOLT_STUN
+            });
+        }
+
+        void HandleOnHit(SpellEffIndex /*effIndex*/)
+        {
+
```
Comment (tkrokli) [inline src/server/scripts/Spells/spell_warrior.cpp]: White space / unneeded blank line

**Comment by tkrokli** [comment]
@HannibalRoG : Looks OK so far, all we need now is for one of the main devs to verify/validate this PR and merge it.

**Comment by wonder2011** [comment]
What about this PR?

**Pull request closed without merging**